// Copyright (c) Huawei Technologies Co., Ltd. 2021-2023. All rights reserved.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;


message NativeHookConfig {
    int32 pid = 1;
    bool save_file = 2;
    string file_name = 3;
    int32 filter_size = 4;
    int32 smb_pages = 5;
    int32 max_stack_depth = 6;
    string process_name = 7;
    bool malloc_disable = 8;
    bool mmap_disable = 9;
    bool free_stack_report = 10;
    bool munmap_stack_report = 11;
    uint32 malloc_free_matching_interval = 12;
    uint32 malloc_free_matching_cnt = 13;
    bool string_compressed = 14;
    bool fp_unwind = 15;
    bool blocked = 16;
    bool record_accurately = 17;
    bool startup_mode = 18;
    bool memtrace_enable = 19;
    bool offline_symbolization = 20;
    bool callframe_compress = 21;
    uint32 statistics_interval = 22;
    string clock = 23;
}
